Formula/rust-analyzer.rb

```diff
@@ -0,0 +1,37 @@
+class RustAnalyzer < Formula
+  desc "Experimental Rust compiler front-end for IDEs"
+  homepage "https://rust-analyzer.github.io/"
+  url "https://github.com/rust-analyzer/rust-analyzer/archive/2020-04-20.tar.gz"
+  sha256 "da47d24f24205c77bd8aeba4f0c2d6b7b12f2462ceab7e19473282c9946bd69c"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked", "--root", prefix, "--path",
+           "crates/rust-analyzer", "--bin", "rust-analyzer"
+  end
+
+  test do
+    require "json"
+
+    initialize_json = JSON.dump(
+      {
+        :id      => 1,
+        :jsonrpc => "2.0",
+        :method  => "initialize",
+        :params  => nil,
+      },
+    )
+
+    input =
+      "Content-Length: #{initialize_json.size + 2}\r\n" \
+      "\r\n" \
+      "#{initialize_json}\r\n"
```

Comment:
Why does it use DOS style newlines?

Reply:
good catch, replaced with UNIX style newlines.